driver/src/metrics/stablex_metrics.rs

```diff
@@ -24,38 +24,44 @@ impl StableXMetrics {
             "dfusion_service_processing_times",
             "timings between different processing stages",
         );
-        let processing_times = IntGaugeVec::new(processing_time_opts, &["stage"]).unwrap();
+        let processing_times =
+            IntGaugeVec::new(processing_time_opts, &[ProcessingStage::LABEL]).unwrap();
```

Comment:
Should we initialize this too?

Reply:
Not sure if it is correct to have a processing time of 0 e.g. for solution submission if it has never actually happened. It could confuse people for metrics where 0s is a reasonable value.

Reply:
I noticed in grafana for solution submission it keeps using the previous value when we don't actually submit a solution. Maybe it would be clearer to reset it to 0 at the start and for every batch where don't really reach the step. Or maybe not, just throwing it out there.

Reply:
Yes, I noticed that too. My concern about clearing is when would we do that? If we clear to early we might not get a sample at the actual value and thus miss the measurement. After all maybe a gauge is not the right metric type here?

Reply:
The sampling problem we already have because technically the next batch could be solved so quickly that the old value cannot be measured even though this is unlikely.

This goes back to the discussion when the metrics were introduced that having these processing times as gauges does not fit the prometheus model. A histogram is more appropriate but then you cannot get the exact processing times and you cannot show just the most recent value.

What we really want is to log the processing times of each stage whenever the stage completes. This could go into a file or sql database table. Then that table contains exactly the information we want. I don't think this accomplishable with Prometheus. Maybe we can log into kibana or some postgres or sql server and have Grafana plot that.

Reply:
Yes, this goes however beyond the scope of this PR. I will create a separate issue for it.